### ISSUE TITLE
[feature request] sending signals to services

### ISSUE THREAD

**Issue by CMB** [davmac314/dinit#134]
It would be nice if I could use dinitctl to signal a service.
Like with runit, I can say `sv hup lighttpd` to tell lighttpd to reload its config.

**Comment by mobin-2008** [comment]
Hi. I thinked about it. Most UNIX-like systems (such as Linux based and BSDs and even Solaris) have `kill` and `pkill`. `pkill` is very easy to use:
````
# for example tell nginx to reload its config
pkill -HUP nginx
````
I don't see a special case to need a special command to send a signal to services because `kill` and `pkill` are exist.

**Comment by CMB** [comment]
Yes kill and pkill will do it, but they may cause collateral damage.
E.G., multiple nginx services running on one system.
The kill and pkill commands operate at the wrong granularity.

Better than pkill would be to ask dinit for the pid, like so:

    kill -HUP "$( dinitctl list |grep ' postgres ' |sed -E 's;^.*pid: ([0-9]+).*;\1;')"

Signaling processes is also racy when you aren't the parent.
There's a race window between the time you obtain the PID and the time you signal the process,
during which the PID might have been recycled.  Granted, the window is pretty tiny, but the kill command is still an act of faith.
With a supervised service, `type=process` in dinit, dinit is the parent, and it can signal the appropriate child in a race-free way.

**Comment by davmac314** [comment]
Note that the usual policy is not to accept feature requests without a patch (see the README) but in this case, yes, it's something that is planned.

**Comment by james-knippes** [comment]
I've implemented signal support in #176 in hopes it may be useful. I would appreciate feedback. I assume the implementation covers your request @CMB ?

**Comment by mobin-2008** [comment]
I forgot to close this issue. This feature has been added in #176 and will be available in 0.17 and later.
